src/store.rs

```diff
@@ -43,12 +43,11 @@ impl Store {
         })
     }
 
-    /// If an output exists for a job, what is it? If we don't have a stored
-    /// output for the job, return `None`.
-    pub fn for_job(&self, key: &job::Key<job::Final>) -> Option<PathBuf> {
-        self.inputs_to_content
-            .get(key)
-            .map(|path| self.root.join(path))
+    pub fn item_for_job(&self, key: &job::Key<job::Final>) -> Result<Option<Item>> {
+        match self.inputs_to_content.get(key) {
+            None => Ok(None),
```

Comment:
Nested Option in Result and returning Ok(None) feels quite odd. Any specific reason why?

Reply:
`Item::from_hex` in the other branch can return an error (e.g. `Item::from_hex("this is not hex".into())`)

So the result is to handle that case, and the option is to say whether things were present or not.

I suppose we could flip them around—an `Option<Result<Item>>`—but that makes life harder on the consuming side for basically the same information.

Is there an easier rust way to say "this operation succeeded, but I have nothing for you?"

Reply:
Ok. Yeah, that is probably the right way to do it unless you want to make one of your errs of the result that you have no data.